### ISSUE TITLE
Can not use prebuilt protobuf of parent project with cmake.

### ISSUE THREAD

**Issue by owent** [open-telemetry/opentelemetry-cpp#740]
**Describe your environment** 
CMake, Windows

**Steps to reproduce**
Having a prebuilt of protobuf.

**What is the expected behavior?**
Do not trigger vcpkg.

**What is the actual behavior?**
Using vcpkg to installl protobuf again.

**Additional context**
Should use `Protobuf_FOUND` instead of `protobuf_FOUND` .

**Comment by ThomsonTan** [comment]
@owent if you are using a prebuilt protobuf, how about gRPC? I think gRPC will be needed for most cases when protobuf is used in this repo.

**Comment by owent** [comment]
> @owent if you are using a prebuilt protobuf, how about gRPC? I think gRPC will be needed for most cases when protobuf is used in this repo.

Using `-DgRPC_PROTOBUF_PROVIDER=package`, and add install path to `CMAKE_PREFIX_PATH` or set `Protobuf_DIR` then we can use our prebuilt protobuf.So does other dependencies of gRPC.

I create a script toolset to build all denpendencies of gRPC.(https://github.com/atframework/cmake-toolset)
And I also build all dependencies of opentelemetry (exclude of ms-gsl, which is include by submodule instead of `find_package`) with it.

**Comment by maxgolov** [comment]
It might be different depending on what version of CMake you use. I've seen it with `Gtest_FOUND` and `gtest_FOUND`, which is different depending on whether you use newer or older package finder. Could it be something like this, can we make both checks?

**Comment by owent** [comment]
> It might be different depending on what version of CMake you use. I've seen it with `Gtest_FOUND` and `gtest_FOUND`, which is different depending on whether you use newer or older package finder. Could it be something like this, can we make both checks?

Thanks, I checked the document of cmake just now and found  `FindProtobuf.cmake` will set `Protobuf_FOUND` from cmake 3.6 and `FindGTest.cmake` will set `GTest_FOUND` from cmake 3.20 in the legacy `MODULE` mode.

I have add codes to check `PROTOBUF_FOUND` and use `Protobuf_PROTOC_EXECUTABLE` to overwrite `PROTOBUF_PROTOC_EXECUTABLE` when it exists. But I also found both `GTEST_FOUND` and `GTest_FOUND` are already checked.

I think the case should be the same as the package name passed into `find_package`, so we should use `GTest_FOUND` instead of `gtest_FOUND`  and it's already the right way.
